sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java

```diff
@@ -485,6 +496,44 @@ protected boolean advanceImpl() throws IOException {
 
   /////////////////////////////////////////////////////////////////////////////
 
+  /** Implementation of {@link #ofProvider}. */
+  public static class OfValueProvider<T> extends PTransform<PBegin, PCollection<T>> {
+    private final ValueProvider<T> provider;
+    private final Coder<T> coder;
+
+    private OfValueProvider(ValueProvider<T> provider, Coder<T> coder) {
+      checkNotNull(provider, "provider");
+      checkNotNull(coder, "coder");
+      this.provider = provider;
+      this.coder = coder;
+    }
+
+    @Override
+    public PCollection<T> expand(PBegin input) {
+      if (provider.isAccessible()) {
+        Values<T> values = Create.of(provider.get());
+        return input.apply(values.withCoder(coder));
+      }
+
+      @SuppressWarnings({"unchecked", "rawtypes"})
+      SerializableCoder<ValueProvider<T>> providerCoder =
+          SerializableCoder.of((Class) provider.getClass());
+      return input
+          .apply(Create.of(provider).withCoder(providerCoder))
```

Comment:
why do you need this? The ValueProvider is already serialized in the transform. You can simply Create.of((Void) null) and then expand in the next ParDo.